bpf/kmesh/probes/tcp_probe.h

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Kmesh */
+
+#ifndef __KMESH_BPF_ACCESS_LOG_H__
+#define __KMESH_BPF_ACCESS_LOG_H__
+
+#include "bpf_common.h"
+
+// direction
+enum {
+    INVALID_DIRECTION = 0,
+    INBOUND = 1,
+    OUTBOUND = 2,
+};
+
+struct tcp_probe_info {
+    struct bpf_sock_tuple tuple;
+    __u64 duration; // ns
+    __u64 close_ns;
+    __u32 family;
+    __u32 protocol;
+    __u32 direction;
+    __u32 sent_bytes;
+    __u32 received_bytes;
+    __u32 conn_success;
+    __u32 state;   /* tcp state */
+    __u32 srtt_us; /* smoothed round trip time << 3 in usecs */
+    __u32 rtt_min;
+    __u32 mss_cache;     /* Cached effective mss, not including SACKS */
+    __u32 total_retrans; /* Total retransmits for entire connection */
+    __u32 segs_in;       /* RFC4898 tcpEStatsPerfSegsIn
+                          * total number of segments in.
+                          */
+    __u32 segs_out;      /* RFC4898 tcpEStatsPerfSegsOut
+                          * The total number of segments sent.
+                          */
+    __u32 lost_out;      /* Lost packets			*/
+};
+
+struct {
+    __uint(type, BPF_MAP_TYPE_RINGBUF);
+    __uint(max_entries, RINGBUF_SIZE);
+} map_of_tcp_info SEC(".maps");
+
+static inline void constuct_tuple(struct bpf_sock *sk, struct bpf_sock_tuple *tuple)
+{
+    if (sk->family == AF_INET) {
+        tuple->ipv4.saddr = sk->src_ip4;
+        tuple->ipv4.daddr = sk->dst_ip4;
+        tuple->ipv4.sport = sk->src_port;
+        tuple->ipv4.dport = bpf_ntohs(sk->dst_port);
+    } else {
+        bpf_memcpy(tuple->ipv6.saddr, sk->src_ip6, IPV6_ADDR_LEN);
+        bpf_memcpy(tuple->ipv6.daddr, sk->dst_ip6, IPV6_ADDR_LEN);
```

Comment:
On the service side, the ipv4 address is reported as ipv6.
Can be modified with reference to #648